### PR TITLE
Fix some impact to past_present_shared_buffer after flash attention

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

**Comment by tianleiwu** [comment]
Please add a test case. It seems that previous tests cannot catch this bug.

File: onnxruntime/contrib_ops/cuda/bert/attention_impl.cu [hunk 1]
```diff
@@ -273,7 +274,7 @@ Status QkvToContext(
     } else {
       // For fused TRT attention, transpose qkv to BxSxNx3xH. For fused causal kernel, transpose to 3xBxNxSxH.
       const int format = (use_fused_kernel ? 2 : 1);
-      T* qkv_add_bias = use_fused_causal ? data.gemm_buffer : nullptr;
+      qkv_add_bias = use_fused_causal ? data.gemm_buffer : nullptr;
       int matrix_to_transpose = ((!use_fused_kernel && past_present_share_buffer) ? 1 : 3);
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cuda/bert/attention_impl.cu:278]: !use_fused_kernel => !use_fused_causal?

Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cuda/bert/attention_impl.cu:278]: We need to add gemm_buffer (all q k v) with bias when causal as they are all needed for the fused kernel.
Definetly there is no need to add kv bias when causal, yet this is the current logic. And this is bug fix only, not to improve performance here. We could improve it in future PR.

**Review by tianleiwu — DISMISSED** [review]

**Review by yufenglee — APPROVED** [review]

**Review by tianleiwu — APPROVED** [review]

**Comment by zhanghuanrong** [comment]
Logic also covered in cross attention PR. Close it.

**Pull request closed without merging**